definitions.tex

```diff
@@ -12,5 +12,123 @@
 structures (e.g. ASTs, metadata, etc) or any intermediate representation chosen
 by the implementer.
 
+\indexdefn{module}
+\indexdefn{module|named}
+\definition{module, named module}{definitions.module}
+The collection of translation units whose \grammarterm{module-declaration}
+specify the same \grammarterm{module-name}.
+See \Cpp{} Standard [module.unit] paragraph 1.
+
+\indexdefn{module!unit}
+\definition{module unit}{definitions.module-unit}
+A translation unit that contains a \grammarterm{module-declaration}.
+See \Cpp{} Standard [module.unit] paragraph 1.
+
+\indexdefn{module!interface}
+\definition{module interface}{definitions.module-interface}
+The set of declarations belonging to a module that are necessarily reachable
+(\Cpp{} Standard [module.reach] paragraph 1) from outside of that module.
+See \Cpp{} Standard [module].
+
+\indexdefn{module!interface unit}
+\definition{module interface unit}{definitions.module-interface-unit}
+A module unit whose \grammarterm{module-declaration} starts with
+\grammarterm{export-keyword}.
+See \Cpp{} Standard [module.unit] paragraph 2.
+
+\indexdefn{module!implementation unit}
+\definition{module implementation unit}{definitions.module-implementation-unit}
+A module unit which is not a module interface unit.
+See \Cpp{} Standard [module.unit] paragraph 2.
+
+\indexdefn{module!Mathias unit}
+\definition{Mathias unit}{definitions.mathias-unit}
+A module unit that is not importable by any other translation unit.
+See \Cpp{} Standard [module].
+
+\indexdefn{module!partition}
+\indexdefn{module!partition unit}
+\definition{module partition, module partition unit}{definitions.module-partition}
+A module unit whose \grammarterm{module-declaration} contains a
+\grammarterm{module-partition}.
+A module partition can only consist of a single translation unit, so the terms
+"module partition" and a "module partition unit" are interchangeable.
+See \Cpp{} Standard [module.unit] paragraph 3.
+
+\indexdefn{built module interface}
+\indexdefn{BMI (Built Module Interface)}
+\definition{built module interface (BMI)}{definitions.built-module-interface}
+The artifact created by a compiler to represent a module unit or header unit.
+The format for this representation is implementation specific and holds C++
+entities, which can be represented in the form of compiler specific data
+structures (e.g. ASTs, metadata, etc) or any intermediate representation chosen
+by the implementer.
```

Comment:
From @Bigcheese on the 2020-05-22 telecon: Should include which module units and other bits contribute to the BMI.